reflex/state.py

```diff
@@ -461,6 +520,10 @@ def __init_subclass__(cls, mixin: bool = False, **kwargs):
         if mixin:
             return
 
+        # Generate a minified state name by converting state count to string
+        if not cls._state_name:
```

Comment:
Seems like we need something like this

```python
if not cls._state_name or cls._state_name == cls.get_parent_state()._state_name:
```

Otherwise once we minify a parent state name `cls._state_name` seems to get the parent's minified name... maybe i'm missing something though.

With the suggested change, i'm actually able to get past the reflex-web error that i mentioned earlier.

Reply:
Thanks for digging into this. You are absolutely right, and I like your suggested fix.

Reply:
Sadly, your solution causes issues with some tests. In fact, just calling `cls.get_parent_state()` in the init_subclass hook causes some tests to fail.  
I adjusted it to `not cls._state_name or cls._state_name in all_state_names`.